### PR TITLE
Reverse a string in Java

### PR DESCRIPTION
**Pull request by shahina-bano**
I have added the solution for the issue #611 . Kindly review it @raksha009 @shreyanspoddar

### CODE REVIEW

File: Code/Java/reverse_string.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
+/*Reverse string
```
Comment (raksha009) [inline Code/Java/reverse_string.java:1]:
```suggestion
/*Reverse a string using 2 pointer Approach

```

File: Code/Java/reverse_string.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import java.util.*;
+import java.lang.*;
+import java.io.*;
+
+class StringReverse{
+    // function for reversing the given string
+	public static void ReverseTheString(){
+	    
+        Scanner sc = new Scanner(System.in); 
+        String input = sc.nextLine();       //taking input string from user
```
Comment (raksha009) [inline Code/Java/reverse_string.java:15]:
```suggestion
        String input = sc.nextLine();       

```
Remove this comment

File: Code/Java/reverse_string.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        int k = input.length();                      //size of the string
```
Comment (raksha009) [inline Code/Java/reverse_string.java:16]: Add comments above the line

File: Code/Java/reverse_string.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        char[] output = input.toCharArray();         //converting it into array to access each Character of string
+        /*using two pointer type approach to swap the characters of string
```
Comment (raksha009) [inline Code/Java/reverse_string.java:18]: Use single line comments in between the code

File: Code/Java/reverse_string.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+Test Case 2:
+Input -
+reversemalayalam
+Output -
+Reversed string: malayalamesrever
+
+Time Complexity: O(k), for traversing the array of characters
+Space Complexity: O(1), no extra space is used
+*/
```
Comment (raksha009) [inline Code/Java/reverse_string.java:53]: Add a new line here

**Review by raksha009 — CHANGES_REQUESTED** [review]

File: Code/Java/reverse_string.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
+/*Reverse a string using 2 pointer Approach
+
+Problem Statement:  Reverse the given string from user.
+                    For e.g initial string: Hello, reversed string: olleH
```
Comment (raksha009) [inline Code/Java/reverse_string.java:4]: Add description in a bit more detail.

File: Code/Java/reverse_string.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+import java.util.*;
+import java.lang.*;
+import java.io.*;
+
+class StringReverse{
+    // function for reversing the given string
+	public static void ReverseTheString(){
+	    
+        Scanner sc = new Scanner(System.in); 
+	//input the string from user
```
Comment (raksha009) [inline Code/Java/reverse_string.java:16]: Remove this line

File: Code/Java/reverse_string.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+        String input = sc.nextLine();       
+	//size of the string
+        int k = input.length();                      
+        char[] output = input.toCharArray();         //converting it into array to access each Character of string
```
Comment (raksha009) [inline Code/Java/reverse_string.java:20]: Add comments above the line throughout the code,

File: Code/Java/reverse_string.java [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+Output -
+Reversed string: ionahforewot
+
+Test Case 2:
+Input -
+reversemalayalam
+Output -
+Reversed string: malayalamesrever
+
+Time Complexity: O(k), for traversing the array of characters
```
Comment (raksha009) [inline Code/Java/reverse_string.java:54]:
```suggestion
Time Complexity: O(k), for traversing the array of characters

```
Specify what is k here

**Review by raksha009 — CHANGES_REQUESTED** [review]

File: Code/Java/reverse_string.java [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+
+import java.util.*;
+import java.lang.*;
+import java.io.*;
+
+class StringReverse{
+    // function for reversing the given string
+	public static void ReverseTheString(){
+        Scanner sc = new Scanner(System.in); 
+	//input the string from user
```
Comment (raksha009) [inline Code/Java/reverse_string.java:17]: Remove this comment on line 17 :)

File: Code/Java/reverse_string.java [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+        String input = sc.nextLine();       
+	//size of the string
+        int k = input.length();
+	//converting it into array to access each Character of string
+        char[] output = input.toCharArray();         
+        //using two pointer type approach to swap the characters of string
```
Comment (raksha009) [inline Code/Java/reverse_string.java:23]: Add a new line before line 23

File: Code/Java/reverse_string.java [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+        char[] output = input.toCharArray();         
+        //using two pointer type approach to swap the characters of string 
+        //one pointer points to starting and other to ending
+        for(int itr = 0, ptr = k-1; itr < ptr; itr++, ptr--){
+	    //swapping the characters
+            char flag = output[itr];                
+            output[itr] = output[ptr];
+            output[ptr] = flag;
+        }
+        System.out.print("Reversed string: ");
```
Comment (raksha009) [inline Code/Java/reverse_string.java:31]: Add a new line before line 31

File: Code/Java/reverse_string.java [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+        // printing the reversed string
```
Comment (raksha009) [inline Code/Java/reverse_string.java:32]: Remove this comment

File: Code/Java/reverse_string.java [hunk 14]
```diff
@@ -0,0 +1,60 @@
+/*Reverse a string using 2 pointer Approach
```
Comment (raksha009) [inline Code/Java/reverse_string.java:1]:
```suggestion
/*
Reverse a string using 2 pointer Approach

```

**Review by raksha009 — CHANGES_REQUESTED** [review]

**Comment by shahina-bano** [comment]
Done :) @raksha009 let me know if anything else you want me to change :)

File: Code/Java/reverse_string.java [hunk 15]
```diff
@@ -0,0 +1,60 @@
...
+                    For e.g initial string: hello, reversed string: olleh.
+		    Using two pointer approach, firstly 'h' and 'o' are swapped similarly other characters follow the same sequence.
+		    In this way, the string is reversed without using extra space.
+*/
+
+import java.util.*;
+import java.lang.*;
+import java.io.*;
+
+class StringReverse{
```
Comment (raksha009) [inline Code/Java/reverse_string.java:14]: class name should be same as file name

Comment (raksha009) [reply Code/Java/reverse_string.java:14]: And add a new line here

File: Code/Java/reverse_string.java [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+    // function for reversing the given string
+	public static void ReverseTheString(){
+        Scanner sc = new Scanner(System.in);
+        String input = sc.nextLine();       
+	//size of the string
```
Comment (raksha009) [inline Code/Java/reverse_string.java:19]: Remove this comment.

File: Code/Java/reverse_string.java [hunk 17]
```diff
@@ -0,0 +1,60 @@
...
+        int k = input.length();
+	//converting it into array to access each Character of string
+        char[] output = input.toCharArray();        
+		
+        //using two pointer type approach to swap the characters of string 
+        //one pointer points to starting and other to ending
+        for(int itr = 0, ptr = k-1; itr < ptr; itr++, ptr--){
+	    //swapping the characters
```
Comment (raksha009) [inline Code/Java/reverse_string.java:27]: Remove this comment.

File: Code/Java/reverse_string.java [hunk 18]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    public static void main (String[] args) throws java.lang.Exception{
+	//calling for the reverse function from main
+        ReverseTheString();         
+    }
+}
+
+/*
+Test Case 1:
```
Comment (raksha009) [inline Code/Java/reverse_string.java:46]: Follow
Test Cases
Input
Output

**Review by raksha009 — CHANGES_REQUESTED** [review]

**Comment by shahina-bano** [comment]
done with the changes @raksha009

File: Code/Java/reverse_string.java [hunk 19]
```diff
@@ -0,0 +1,60 @@
...
+import java.io.*;
+
+class reverse_string{
+	
+   	// function for reversing the given string
+	public static void ReverseTheString(){
+        Scanner sc = new Scanner(System.in);
+        String input = sc.nextLine();       
+        int k = input.length();
+	//converting it into array to access each Character of string
```
Comment (raksha009) [inline Code/Java/reverse_string.java:21]: Indentation

File: string/readme.md [hunk 20]
```diff
@@ -13,4 +13,5 @@ A string is a class that contains a char array, but automatically manages it for
 
 ## Questions :
 * Revere a string using Stack ---->[C++](/Code/C++/reverse_a_string_using_stack.cpp)
+* Reverse a string ---->[Java](/Code/Java/reverse_string.java)
 * Reverse a string ---->[C++](/Code/C++/reverse_string.cpp)
```
Comment (raksha009) [inline string/readme.md:17]: Edit 17th line 
```suggestion
* Reverse a string ---->[C++](/Code/C++/reverse_string.cpp) | [Java](/Code/Java/reverse_string.java)
```

**Review by raksha009 — CHANGES_REQUESTED** [review]

File: Code/Java/reverse_string.java [hunk 21]
```diff
@@ -0,0 +1,60 @@
...
+        }
+		
+        System.out.print("Reversed string: ");
+        for(int itr = 0; itr < k; itr++){
+            System.out.print(output[itr]);
+        }
+    }
+
+    public static void main (String[] args) throws java.lang.Exception{
+	//calling for the reverse function from main
```
Comment (raksha009) [inline Code/Java/reverse_string.java:39]: plz correct Indentation here too

**Review by raksha009 — CHANGES_REQUESTED** [review]

**Review by raksha009 — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**